### PR TITLE
Fix min k8s version supported

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**
Based on https://istio.io/latest/docs/releases/supported-releases/#support-status-of-istio-releases.
Officially supported k8s version from 1.21-1.24

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [x] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [x] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: istioctl/pkg/install/k8sversion/version.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ import (
 const (
 	// MinK8SVersion is the minimum k8s version required to run this version of Istio
 	// https://istio.io/docs/setup/platform-setup/
-	MinK8SVersion               = 19
+	MinK8SVersion               = 21
```
Comment (hzxuzhonghu) [inline istioctl/pkg/install/k8sversion/version.go:31]: We donot need to upgrade this unless it does not work in 1.19 or 1.20 anymore

Comment (hanxiaop) [reply istioctl/pkg/install/k8sversion/version.go:31]: @hzxuzhonghu Shouldn't this be the value matches https://preliminary.istio.io/latest/docs/releases/supported-releases/#support-status-of-istio-releases, and https://github.com/istio/istio.io/blob/4827604b9c90e272d0d25bb173f79e19b2b57b3d/data/args.yml#L31-L32? I've checked the old PRs, and the min k8s version was always updated along with the Istio version.

Comment (hzxuzhonghu) [reply istioctl/pkg/install/k8sversion/version.go:31]: This is not to endusers, so i think not a must,  it is ok to be a superset of our announced supported versions.

**Pull request closed without merging**